### DISCUSSION TITLE
How to use GPU when I don't use Nvidia GPU

### DISCUSSION THREAD

**Discussion by rkdtmdals7710** [ultralytics/ultralytics#10076] [Q&A]
In a Raspberry Pi 5 environment using videocore vii gpu, I am running ultralytics/yolov5/detect.py with YOLOv5n custom model weights. However, as a result of cpu and gpu usage queries, memory is assigned only to cpu, and not at all to gpu. How can I allocate gpu to memory when running detect.py without using gpu from Nvidia. Additionally, I tried changing the --device on detect.py , but it was not recognized except for 'cpu'

**Comment by pderrenger** [comment]
Hello!

For environments not supported by CUDA (like Raspberry Pi's VideoCore VI GPU), you cannot directly use GPUs with Ultralytics YOLO models in the same way you would on Nvidia GPUs. The YOLO models primarily leverage CUDA for GPU acceleration, which is specific to Nvidia GPUs.

A potential workaround to utilize the GPU on a Raspberry Pi might involve converting the model to a format compatible with the hardware, such as TensorFlow Lite, which can then run on the Raspberry Pi's GPU through software that supports it, such as TensorFlow Lite's GPU delegate. Here's a brief outline of steps you might consider:

1. Export your YOLOv5 model to ONNX format:
    ```bash
    python models/export.py --weights yolov5n.pt --img 640 --batch 1
    ```
2. Convert the ONNX model to TensorFlow Lite format. Tools like [ONNX-TensorFlow](https://github.com/onnx/onnx-tensorflow) and [TensorFlow Lite Converter](https://www.tensorflow.org/lite/convert) might be useful.
3. Use TensorFlow Lite with GPU delegate on your Raspberry Pi to run the model.

Remember, the performance improvement depends heavily on the Raspberry Pi's hardware capabilities and the optimizations possible through the conversion process. 🚀

If this solution doesn't fit your scenario or if you encounter any specific issues, please feel free to provide more details!

Keep exploring, and I hope this helps you on your project!
